### PR TITLE
fix the bug that the value of gasLimit and gasUsed in block is not correct

### PR DESCRIPTION
**Pull request by lichenxi-webank**

### CODE REVIEW

**Comment by bxq2011hust** [comment]
[CodeFactor](https://www.codefactor.io/repository/github/fisco-bcos/fisco-bcos) found an issue: Complex Code

It's currently on:
[libblockchain\BlockChainImp.cpp
](https://github.com/FISCO-BCOS/FISCO-BCOS/blob/04a9b623ac7f43a30cd6d17103287dd630c0bee0/libblockchain/BlockChainImp.cpp)Commit 04a9b623ac7f43a30cd6d17103287dd630c0bee0

File: libblockchain/BlockChainImp.cpp [hunk 1]
```diff
@@ -331,6 +331,7 @@ bool BlockChainImp::checkAndBuildGenesisBlock(GenesisBlockParam& initParam)
                 SYSTEM_CONFIG_VALUE, boost::lexical_cast<std::string>(initParam.txGasLimit));
             entry2->setField(SYSTEM_CONFIG_ENABLENUM, "0");
             tb->insert(SYSTEM_KEY_TX_GAS_LIMIT, entry2);
+            m_gasLimit = initParam.txGasLimit;
```
Comment (wheatli) [inline libblockchain/BlockChainImp.cpp]: gaslimit should be stetted when the block is created

File: libblockchain/BlockChainImp.cpp [hunk 2]
```diff
@@ -857,6 +858,9 @@ CommitResult BlockChainImp::commitBlock(Block& block, std::shared_ptr<ExecutiveC
         return CommitResult::ERROR_PARENT_HASH;
     }
 
+    block.header().setGasLimit(m_gasLimit);
+    block.header().setGasUsed(block.gasUsed());
```
Comment (wheatli) [inline libblockchain/BlockChainImp.cpp]: gassed should be stetted when finish the execution of block

File: libethcore/Block.h [hunk 3]
```diff
@@ -121,6 +121,17 @@ class Block
     }
     /// get hash of block header
     h256 blockHeaderHash() { return m_blockHeader.hash(); }
+    u256 gasUsed()
+    {
+        if (m_transactionReceipts.size() != 0)
+        {
+            return m_transactionReceipts.back().gasUsed();
```
Comment (wheatli) [inline libethcore/Block.h]: Don't you need to accelerate the gassed of all tx ?

**Pull request closed without merging**